spec/tasks/owasp-dep-check/owasp-dep-check_spec.rb

```diff
@@ -0,0 +1,165 @@
+require 'spec_helper'
+
+require 'glue'
+require 'glue/event'
+require 'glue/tracker'
+require 'glue/tasks'
+require 'glue/tasks/owasp-dep-check'
+
+
+describe Glue::OWASPDependencyCheck do
+
+  DEP_CHECK_TARGETS_PATH = 'spec/tasks/owasp-dep-check/targets'
+
+  def get_dep_check (target = 'nil_target')
+    trigger = Glue::Event.new(target)
+    trigger.path = File.join(DEP_CHECK_TARGETS_PATH, target)
+    tracker = Glue::Tracker.new({})
+    @dep_check_path = tracker.options[:owasp_dep_check_path]
+    Glue::OWASPDependencyCheck.new(trigger, tracker)
+  end
+
+  def get_raw_report (target, subtarget = nil)
+    path = File.join(get_target_path(target, subtarget), "dependency-check-report.xml")
+    File.read(path).chomp
+  end
+
+  def get_target_path (target, subtarget = nil)
+    if subtarget.nil?
+      File.join(DEP_CHECK_TARGETS_PATH, target)
+    else
+      File.join(DEP_CHECK_TARGETS_PATH, target, subtarget)
+    end
+  end
+
+  def cli_args (target, subtarget = nil)
+    target_path = get_target_path(target, subtarget)
+    [true, @dep_check_path, "--project", "Glue", "-f", "ALL", "-out", target_path, "-s", target_path]
+  end
+
+  describe "#initialize" do
+    let(:task) { @task }
+    before(:all) { @task = get_dep_check }
+
+    it "sets the correct 'name'" do
+      expect(task.name).to eq('OWASP Dependency Check')
+    end
+
+    it "sets the correct 'stage'" do
+      expect(task.stage).to eq(:code)
+    end
+
+    it "sets the correct 'labels'" do
+      expect(task.labels).to eq(%w(code java .net).to_set)
+    end
+  end
+
+  describe "#supported?" do
+    subject(:task) { get_dep_check }
+
+    context "when 'runsystem' cannot run the task" do
```

Comment:
Add also a positive task - they the task is supported

Reply:
Positive check for `#supported?` done :)
As for the JAR files, I don't think there's a way to test the dependency-check CLI tool without them, only for sbt, maven and gradle plugins which specify dependencies in a build file, and the JARs are then downloaded during the build process.
I took some care to pick the smallest JARs I could find.